runbot_gitlab/models/runbot_repo.py

```diff
@@ -151,23 +151,25 @@ class RunbotRepo(models.Model):
     @api.model
     def create(self, vals):
         repo_id = super(RunbotRepo, self).create(vals)
-        set_gitlab_ci_conf(
-            vals.get('token'),
-            vals.get('name'),
-            self.domain(),
-            repo_id.id,
-        )
+        if vals.get('uses_gitlab', False):
+            set_gitlab_ci_conf(
+                vals.get('token'),
+                vals.get('name'),
+                self.domain(),
+                repo_id.id,
+            )
         return repo_id
 
     @api.multi
     def write(self, vals):
         super(RunbotRepo, self).write(vals)
-        set_gitlab_ci_conf(
-            vals.get('token', self.token),
-            vals.get('name', self.name),
-            self.domain(),
-            self.id,
-        )
+        if vals.get('uses_gitlab', self.uses_gitlab):
```

Comment:
You should be able to just check `self.uses_gitlab` since vals have already written to super.